spring-multirabbit/src/main/java/org/springframework/amqp/rabbit/annotation/ExtendedMultiRabbitListenerAnnotationBeanPostProcessor.java

```diff
@@ -23,9 +26,15 @@
  * @author Wander Costa
  * @see RabbitListenerAnnotationBeanPostProcessor
  */
+@Component
 public final class ExtendedMultiRabbitListenerAnnotationBeanPostProcessor
         extends RabbitListenerAnnotationBeanPostProcessor {
 
+    public ExtendedMultiRabbitListenerAnnotationBeanPostProcessor(
+        final ConnectionFactory multiRabbitConnectionFactory) {
```

Comment:
I think this is the main trick to solve the issue of the admins not found. As it requires the `ConnectionFactory`, it will require to process `MultiRabbitAutoConfiguration` before.
Maybe you can try annotating the class with `@DependsOn(MultiRabbitConstants.CONNECTION_FACTORY_BEAN_NAME)` to avoid the in-class logic dependency.
If it doesn't work, we can keep this current implementation, but need to call `super()` for compatibility with non-multirabbit.

Reply:
Actually, it will not fix the issue with the admins and creation of queues...(next PR will 😄)
Yes, I've tried with `@DependsOn...` but didn't have any effect, so I used the constructor injection. 
I didn't get the call to `super()`, is it really required? 
BTW, `ExtendedMultiRabbitListenerAnnotationBeanPostProcessor` will be created only if multirabbit is enabled and we will keep the import to `RabbitAutoConfiguration`... let me know if I'm missing something.

Reply:
Ahh ok, thx for clarifying about `DependsOn`...
Regarding the `super()`, it's about the compatibility with original spring implementation. It was missing before, but as you introduced the constructor, it might worth it adding it.

Reply:
Ok, I got it.
Added call to super 👍🏻